src/test_application/main.cpp

```diff
@@ -389,6 +410,63 @@ void load_a_bunch_of_stuff()
     OSP_LOG_INFO("Resource loading complete");
 }
 
+static void resource_for_each_type(osp::ResTypeId type, void(*do_thing)(osp::ResId))
+{
+    lgrn::IdRegistry<osp::ResId> const &rReg = g_resources.ids(type);
+    for (std::size_t i = 0; i < rReg.capacity(); ++i)
+    {
+        if (rReg.exists(osp::ResId(i)))
+        {
+            do_thing(osp::ResId(i));
+        }
+    }
+}
+
+void clear_resource_owners()
+{
+    using namespace osp::restypes;
+
+    // Texture resources contain osp::TextureImgSource, which refererence counts
+    // their associated image data
+    resource_for_each_type(gc_texture, [] (osp::ResId id)
+    {
+        auto *pData = g_resources
+                .data_try_get<osp::TextureImgSource>(gc_texture, id);
+        if (pData == nullptr)
+        {
+            return;
```

Comment:
minor: I recommend doing this in reverse.

if(pData != nullptr)
{
    call the function
}
// else nothing